push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/FCMPushNotificationSender.java

```diff
@@ -198,25 +205,27 @@ private void processFirebaseRegularMessage(List<String> pushTargets, Builder fir
         }
     }
 
-    private void processFirebaseMultiCastMessage(List<String> pushTargets, MulticastMessage.Builder firebaseMulticastMessage, NotificationSenderCallback callback) {
+    private void processFirebaseMultiCastMessage(List<String> pushTargets, MulticastMessage.Builder firebaseMulticastMessage, NotificationSenderCallback callback, AndroidVariant androidVariant) {
         try {
             logger.info(String.format("Sent push notification to FCM Server for %d registrationIDs", pushTargets.size()));
             firebaseMulticastMessage.addAllTokens(pushTargets);
             BatchResponse multicastResult = firebaseMessaging.sendMulticast(firebaseMulticastMessage.build());
             // after sending, let's identify the inactive/invalid registrationIDs and trigger their deletion:
-            handleMulticastResponses(multicastResult, pushTargets);
+            handleMulticastResponses(multicastResult, pushTargets, androidVariant);
             callback.onSuccess();
         } catch (Exception e) {
             callback.onError(String.format("Error sending payload to FCM server: %s", e.getMessage()));
         }
     }
 
-    private void handleMulticastResponses(BatchResponse multicastResult, List<String> registrationIDs) {
+
+    private void handleMulticastResponses(BatchResponse multicastResult, List<String> registrationIDs, AndroidVariant androidVariant) {
         // get the FCM send responses for all the client devices:
         final List<SendResponse> responses = multicastResult.getResponses();
 
         // read the responses:
         int i = 0;
+        Set<String> inactiveTokens = new HashSet<>();
         for (SendResponse response : responses) {
             // use the current index to access the individual responses
             FirebaseMessagingException exception = response.getException();
```

Comment:
u can use `SendResponse.isSuccessful()`
its a new method from the new api that saves the 2 ifs:
Returns whether the send operation was successful or not. When this method returns true, getMessageId() is guaranteed to return a non-null value. When this method returns false getException() is guaranteed to return a non-null value.
Returns:
A boolean indicating success of the operation.